### ISSUE TITLE
Shadow effect clips

### ISSUE THREAD

**Issue by polarathene** [psifidotos/Latte-Dock#817]
If the shadow effect size is too big it will clip off(alpha/mask) from the dock/panel area. This is not an issue when interacting with the dock where the shadows draw correctly. In idle they will clip again.

Side note, since the settings dialog doesn't appear to be the usual pop up window dialog for settings, and the extra visual changes it applies, it's kind of difficult to tweak settings like shadow and know what they look like without closing the settings window.

**Comment by psifidotos** [comment]
Screenshot please

**Comment by polarathene** [comment]
Opacity/Size

100/100
![latte_100_100_no_scale](https://user-images.githubusercontent.com/5098581/34430129-647810bc-ecc6-11e7-8501-10a47935b9db.png)

100/100, scaling(notice that shadows are fine, the error is still viewable on largest scaled item, FireFox a bit more noticeable, not so much with the folder icon, it's shadow is clipped on the rightside, rocket icon also has some scaling quality issues, barely noticeable here)
![latte_shadow_100_100_scaled](https://user-images.githubusercontent.com/5098581/34430140-aab62848-ecc6-11e7-98ac-0f90afb4ff35.png)

60/100, scaling
![latte_shadow_100_scale](https://user-images.githubusercontent.com/5098581/34430196-4dc742b0-ecc7-11e7-8f53-d9a195869253.png)

100/100, shadows disabled. Note only the Rocket(app launcher plasmoid) doesn't apply shadows anymore, icons only applies regardless.
![latte_100_100_disabled](https://user-images.githubusercontent.com/5098581/34430201-669c2468-ecc7-11e7-911d-8dd982106314.png)

100/0, note that size does not affect shadows on the Rocket icon/plasmoid, it remains consistent despite adjusting size, opacity works as expected however.
![latte_100_0](https://user-images.githubusercontent.com/5098581/34430211-907c353e-ecc7-11e7-82f3-edcf35588c80.png)

60/30, I settled on this, size is much smaller and opacity is low enough that you don't get clipping.
![latte_shadow_60_30](https://user-images.githubusercontent.com/5098581/34430260-f5244292-ecc7-11e7-875e-87479e3f1cf7.png)

---

I noticed similar issue with Blur effect(edges have sharp clipped effect, even if use background panel with border, it is blurred out?

**Comment by psifidotos** [comment]
Please dont mess issues .... keep this as the shadow issue... in the shadow issue you mentioned three different issues, I will look at them : applets shadows arent consistent with icon  taskmanager... thus is weird... very big shadows ate clipped in normal mode both at thickness amd length edges...

rocket icon has some scaling issues this is irrelevant and probably nothing related to Latte.

Blur clipping open a different issue and attach screenshots to understand

**Comment by psifidotos** [comment]
>  Note only the Rocket(app launcher plasmoid) doesn't apply shadows anymore, icons only applies regardless.

I cant reproduce the issue with the shadow to not be identical to all items in the dock... 

1. The rocket which plasmoid is it?
2. Add more plasmoids in the dock dont you get the same shadow?

**Comment by psifidotos** [comment]
@polarathene feel free to open new issues if something hasnt been fixed...

![default](https://user-images.githubusercontent.com/2216474/34481967-bec330e8-efbd-11e7-9406-12d3aa707e8f.png)

**Comment by polarathene** [comment]
@psifidotos The rocket was the Application Launcher plasmoid. 

On my modern system i5-6500 with nvidia GTX 1070 graphics non-free I see the shadow issue I talk about doesn't have any clipping until I move mouse away from the dock(so all icons are same size), just as it begins auto-hide, the shadows all clip and it animates away. I have just installed 0.7.2 via AUR package on Manjaro.

Perhaps the problem is due to themes in some way or the dated hardware(core 2 duo dual core laptop with ATI radeon graphics) that I reported this issue from? Blur seems to look fine on my personal system, unlike the old laptop. This might be why you are unable to reproduce?

As soon as I have access to the laptop again I'll test additional plasmoids for you to answer your 2nd question.

**Comment by psifidotos** [comment]
@polarathene 

> The rocket was the Application Launcher plasmoid.

which one? there are plenty... 

>  I have just installed 0.7.2 via AUR package on Manjaro.

the patch was only at the master version. I jave just merged it into the stable branch also

> Perhaps the problem is due to themes in some way or the dated hardware(core 2 duo dual core laptop with ATI radeon graphics) that I reported this issue from?

It wasnt. Latte wasnt taking into account the user set shadow in order to calculate the mask boundaries.

> Blur seems to look fine on my personal system, unlike the old laptop. This might be why you are unable to reproduce?

this must be either a specific plasma theme issue or this machine isnt updated. There were some relevant issues the last summer but it was kde frameworks fault which was resolved.

**Comment by polarathene** [comment]
> which one? there are plenty...

The one that says `Application Launcher`... is only 3 default launchers with KDE install. Application `Dashboard`, `Launcher` and `Menu`.

> the patch was only at the master version. I jave just merged it into the stable branch also

I didn't mean to say I had installed your latest commit, just what version I was using on personal system(I noticed I did not mention it in my opening issue).

> It wasnt. Latte wasnt taking into account the user set shadow in order to calculate the mask boundaries.

The graphical issue while similar on personal system are different than what I experienced on the laptop. The shadow clip issues were more problematic/noticeable, on personal system I did not notice as much and thought only affected in idle/transition state not hover. I looked again, was not as noticeable perhaps due to larger resolution.

It does not have the shadow issue that the laptop did with launcher plasmoid settings. I don't notice issue with blur either. 

> this must be either a specific plasma theme issue or this machine isnt updated. There were some relevant issues the last summer but it was kde frameworks fault which was resolved.

I will have to confirm if it is theme when I have access to laptop next, but I think it is possible due to old hardware it could be the radeon ATI graphic drivers? It runs a very recent Manjaro release, I installed it onto the system not long before raising this issue.

I have noticed that Breeze Light/Dark widgets do not look correct theme as on my personal system. I'm not sure what caused this, I have downloaded the themes or themes based on Breeze, yet all have the different widget theme style(even preview thumbnail in system settings looks different).
